### ISSUE TITLE
ReactiveSwift 4.0: map(value: ) overload takes precedence over map(_ transform: )

### ISSUE THREAD

**Issue by nalexn** [ReactiveCocoa/ReactiveSwift#669]
We're migrating our code base from ReactiveSwift 3.x to 4.0 right now, and Xcode generates a lot of compilation errors for the reason that can be narrowed down to the following example:

``` swift
    let doubleValue = Property<Double>(value: 123.0)
    let isGreaterThanZero = doubleValue.map { $0 > 0.0 }
    let isGreaterThanZero2 = doubleValue.map({ $0 > 0.0 })
```

In the example above, you'd expect both `isGreaterThanZero` and `isGreaterThanZero2` to be typed as `Property<Bool>`, however, this is only true for `isGreaterThanZero2`.

The type for `isGreaterThanZero` is `Property<(Double) -> Bool>`, which is the result of `map(value: )` overload taking precedence over `map(_ transform: )`

This is not happening in ReactiveSwift 3.x

**Comment by mdiep** [comment]
What version of Swift are you using?

More context here: https://github.com/ReactiveCocoa/ReactiveSwift/pull/604

**Comment by mdiep** [comment]
Note that the underlying issue, https://bugs.swift.org/browse/SR-6989, has been address in Swift. I'm not sure if it's in 4.1, but it will be fixed in 4.2.

**Comment by nalexn** [comment]
I can confirm we're using Swift 4.1. Waiting for the public release of Swift 4.2 then. In the meanwhile, had to introduce a temporary replacement for the `map`:

``` swift
public extension PropertyProtocol {
  func mapTo<U>(_ transform: @escaping (Self.Value) -> U) -> ReactiveSwift.Property<U> {
    return map(transform)
  }
}
```

**Comment by ABeard89** [comment]
@mdiep The Swift bug is still open, so I guess this wasn't fixed for 4.2?

While we wait for a Swift fix, is there a recommended workaround?

**Comment by mdiep** [comment]
> The Swift bug is still open, so I guess this wasn't fixed for 4.2?

The bug has been fixed. The ticket remains open because it's tracking the suggestion that trailing closure syntax should be disabled entirely for functions that don't explicitly take a function.

**Comment by ABeard89** [comment]
> > The Swift bug is still open, so I guess this wasn't fixed for 4.2?
> 
> The bug has been fixed. The ticket remains open because it's tracking the suggestion that trailing closure syntax should be disabled entirely for functions that don't explicitly take a function.

Oh, I thought I still had the problem when I tested with 4.2. My bad!

**Comment by RuiAAPeres** [comment]
Hello. 👋  Thanks for opening this issue. Due to inactivity, we will soft close the issue. If you feel that it should remain open, please let us know. 😄
